client/src/App.js

```diff
@@ -38,48 +38,36 @@ const appStyles = makeStyles((theme) => ({
 }));
 
 function App() {
-  const [loggedIn, setLoggedIn] = useState(false);
+  //Update is literally only here to force a rerender of App when user signs up or signs in
+  const [update, setUpdate] = useState(false);
```

Comment:
Ah. This is the same as using [forceUpdate](https://reactjs.org/docs/react-component.html#forceupdate) which is a React antipattern and should be avoided.

The original method that was deleted, was not the best way of handling things either though.  
One way to handle this quickly for right now:
1. Let's keep the `const [loggedIn, setLoggedIn]` state. 
2. Pass `setLoggedIn` as a prop to `SignUp` and `SignIn` 
3. In both components, for example, on [this line](https://github.com/hatchways/team-kim-possible/blob/b3eda06389be59e911a399d8b5e5b3e0a7fc5290/client/src/components/UserAccount/SignIn.js#L59), after adding to localstorage, also `setLoggedIn(true)`

This way we can remove the constantly checking `useEffect`

Reply:
Yup, found a bug in dev so will fix that before editing/merging anything. I am also planning on using something like const [loggedIn, setLoggedIn] = useState(Boolean(sessionStorage.loggedIn)) to avoid having to re-login everytime someone refreshes the page and reloads the app.